### DISCUSSION TITLE
How does autobrr determine TorrentLeech's release size from IRC annoucement?

### DISCUSSION THREAD

**Discussion by anhtule88** [autobrr/autobrr#1453] [Q&A] [answered]
The IRC messages do not contain torrent size. How does autobrr get the release size?

**Comment by anhtule88** [comment]
Oh, I need to input RSS key when adding TorrentLeech indexer. My bad.

**Reply by zze0s** [reply, marked as answer]
The RSS key is used for grabbing torrents without cookies, which makes things a lot easier.

It's explained here how it's handled when size is not announced https://autobrr.com/filters#rules 

Generally it will download and parse the .torrent file to get that value and in other cases like for RED/OPS, GGn, PTP and BTN it will use their API to fetch it because they keep track of downloads that are not started as well as API requests using way less resources on both their servers and autobrr.

**Comment by anhtule88** [comment]
Recently I got a lot of 0 bytes torrents but there are still some few torrents with size. How can I debug what is the reason?

For more information, those 0 bytes torrents are matched to a filter without download action (but this filter still has a size related condition). Those which matched download enabled filters are getting correct torrent sizes. Previously, all releases are getting sizes, regardless of the available of download action.

**Reply by zze0s** [reply]
Those might be rejected earlier so they don't try to check size and thus end up as 0 bytes in the releases list.
